todo.sh

```diff
@@ -1120,6 +1126,29 @@ case $action in
     IFS=$SAVEIFS
     ;;
 
+"addp")
+    errmsg="usage: $TODO_SH addp PRIORITY \"TODO ITEM\""
+    [ -z "$2" ] && die "$errmsg"
+    priority="$2"
+    [ -z "$3" ] && die "$errmsg"
+    
+    # Validate priority is a single letter A-Z
+    if ! echo "$priority" | grep -q '^[A-Za-z]$'; then
```

Comment:
We already have modern Bash-style checks ([here](https://github.com/todotxt/todo.txt-cli/blob/564ed3373de9495ee4f654879617ce4bfa322507/todo.sh?plain=1#L1420)); no need to use `grep`.

```suggestion
    if [[ "$priority" != @([A-Za-z]) ]]; then
```